output/source-info-v2.ts

```diff
@@ -103,6 +128,15 @@ const D2Sources: {
     sourceHashes: [
       2952071500, // Source: Into the Light
     ],
+    missingSource: [
+      211732170, // Hammerhead
+      243425374, // Falling Guillotine
+      570866107, // Succession
+      2228325504, // Edge Transit
+      2499720827, // Midnight Coup
+      3757612024, // Luna's Howl
+      3851176026, // Elsie's Rifle
+    ],
```

Comment:
I'm not sure I understand the difference in intent or usage between the `missingSource` list, and the `itemHashes` list. We only use these properties for one thing, which is checking to see if an item is from a given source:

(this is simplified:
```ts
return (item) =>
  D2Sources[filterValue].sourceHashes?.includes(item.source) ||
  D2Sources[filterValue].itemHashes?.includes(item.hash) ||
  missingSources[filterValue]?.includes(item.hash);
```

Since we check `D2Sources[filterValue].itemHashes`, then `missingSources[filterValue]` (which after this PR would be `D2Sources[filterValue].missingSource`), it seems like we could combine the two lists into a single `itemHashes` list (with duplicates removed). Furthermore, we could omit the hash of any item whose source already shows up in `sourceHashes`.

As an example, the 30th anniversary missing sources includes `286271818, // Twisting Echo Cloak`, which DIM is able to find a collectible for through a name match in the collectibles presentation nodes (https://data.destinysets.com/i/Collectible:1103944132) which has a source hash of 675740011, which is already in the list of 30th anniversary source hashes. So in practice, DIM won't consult that list of missing sources for this item.

Reply:
yes the name match is what missing-source-info was used for...
![image](https://github.com/DestinyItemManager/d2-additional-info/assets/4798491/aaa49c2f-cf2e-4b9d-a68d-73d4e329a089)

Reply:
This is the code that finds matching collectibles for items, by name:
https://github.com/DestinyItemManager/DIM/blob/master/src/app/records/collectible-matching.ts

Reply:
I'm guessing it didnt exist back https://github.com/DestinyItemManager/d2-additional-info/pull/153

Reply:
Yeah. So my suspicion is that missing-source-info is actually useless to DIM, since it already figures that out at runtime.